### DISCUSSION TITLE
Add support for emitting custom events in custom layouts

### DISCUSSION THREAD

**Discussion by MajedMuhammad** [directus/directus#14973] [Feature Requests]
### Describe the Bug

LayoutConfig interface doesn't support custom emit events definition. Emits should be defined before initialize the setup function otherwise will be "Extraneous non-emits event listeners".

### To Reproduce

Try to use emit with events other than ('update:selection', 'update:layoutOptions', or 'update:layoutQuery').

### What version of Directus are you using?

9.14.2

### What version of Node.js are you using?

16.3.0

### What database are you using?

MySQL

### What browser are you using?

Chrome

### How are you deploying Directus?

locally

**Comment by br41nslug** [comment]
As according to the docs those are the 3 supported emit events (https://docs.directus.io/extensions/layouts/#available-emits) 
I've moved this to a discussion as a feature request/improvement rather than an issue.

**Reply by MajedMuhammad** [reply]
Yes, the new feature can be the ability to add new emit events. It will be very helpful and make it more flexible to use Layouts.

**Reply by rijkvanzanten** [reply]
What would these new events do? The app's parent component will only listen for the above mentioned events, so emitting any custom events from a layout would effectively be a no-op 🤔

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
